pagefaultgames/pokerogue
# [Daily] Daily standardization

This Pull Request makes a few changes to Daily Runs as requested by Vicksin

## What are the changes the user will see?
- Daily runs will now instead have a set luck for all runs, different per day
- Eviolite and Mini Black Hole will always be obtainable in daily runs
- The player starts with a Map in Daily Runs, in addition to their 300% charms bonus

## Why am I making these changes?
The Map allows players to more easily explore biomes and get more out of each day.
Removing the unlock requirement for Mini Black Hole and Eviolite allows players who have unlocked it to get the same items as players who haven’t.
By making luck consistent for all players, and making variants not affect RNG, all players should have the exact same run, even if they encounter Shiny Pokémon.

## What are the changes from a developer perspective?
Eviolite is now always available in daily runs
The function to get the player’s Luck checks if you’re in a daily run, and if you are, it sets it to a different number determined by luck seed
A Map has been inserted in between the Golden EXP Charms and the starters’ held items
Variant selection is placed inside of an ExecuteWithSeedOffset to prevent it from affecting the run’s RNG

### Screenshots/Videos
See below

## How to test the changes?
Open up a Daily Run.
- You will have a Map in your key items
- The shop should have a constant luck stat regardless of your shiny count.

## Checklist
- [x] [working on] Have I considered writing automated tests for the issue?
